src/main/java/com/csforge/sstable/CassandraUtils.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.dht.IPartitioner;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.metadata.MetadataComponent;
+import org.apache.cassandra.io.sstable.metadata.MetadataType;
+import org.apache.cassandra.io.sstable.metadata.ValidationMetadata;
+import org.apache.cassandra.io.util.FileUtils;
+import org.apache.cassandra.schema.*;
+import org.apache.cassandra.utils.FBUtilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CassandraUtils {
+    private static final Logger logger = LoggerFactory.getLogger(CassandraUtils.class);
+    private static final AtomicInteger cfCounter = new AtomicInteger();
+
+    public static CFMetaData tableFromBestSource(String sstablePath) throws IOException, NoSuchFieldException, IllegalAccessException {
+        // TODO add CQL/Thrift mechanisms as well
+        String cqlPath = System.getProperty("sstabletools.schema", "schema.cql");
+        InputStream in = Query.class.getClassLoader().getResourceAsStream(cqlPath);
+        CFMetaData metadata;
+        if(in == null) {
+            logger.debug("Using metadata from sstable");
+            metadata = CassandraUtils.tableFromSSTable(sstablePath);
+        } else {
+            logger.debug("Using metadata from CQL schema in " + cqlPath);
+            metadata = CassandraUtils.tableFromCQL(in);
+        }
+        return metadata;
+    }
+
+    public static CFMetaData tableFromCQL(InputStream source) throws IOException {
+        String schema = CharStreams.toString(new InputStreamReader(source, "UTF-8"));
+        CFStatement statement = (CFStatement) QueryProcessor.parseStatement(schema);
+        String keyspace = Strings.isNullOrEmpty(statement.keyspace()) ? "turtle" : statement.keyspace();
+        statement.prepareKeyspace(keyspace);
+
+        Schema.instance.setKeyspaceMetadata(KeyspaceMetadata.create(keyspace, KeyspaceParams.local(), Tables.none(),
+                Views.none(), Types.none(), Functions.none()));
+        return ((CreateTableStatement) statement.prepare().statement).getCFMetaData();
+    }
+
+    private static Object readPrivate(Object obj, String name) throws NoSuchFieldException, IllegalAccessException {
+        Field type = obj.getClass().getDeclaredField(name);
+        type.setAccessible(true);
+        return type.get(obj);
+    }
+
+    public static CFMetaData tableFromSSTable(String path) throws IOException, NoSuchFieldException, IllegalAccessException {
+        Descriptor desc = Descriptor.fromFilename(path);
```

Comment:
This is not a new issue but just happened to notice it.  If you provide an sstable file by relative path, I get a NPE here since cassandra doesn't seem to resolve the absolute path:

```
java -jar target/sstable-tools-3.0.0-SNAPSHOT.jar toJson ma-1-big-Data.db 
java.lang.NullPointerException
	at org.apache.cassandra.io.sstable.Descriptor.fromFilename(Descriptor.java:289)
	at org.apache.cassandra.io.sstable.Descriptor.fromFilename(Descriptor.java:204)
	at com.csforge.sstable.CassandraUtils.tableFromSSTable(CassandraUtils.java:71)
	at com.csforge.sstable.CassandraUtils.tableFromBestSource(CassandraUtils.java:45)
	at com.csforge.sstable.SSTable2Json.main(SSTable2Json.java:87)
	at com.csforge.sstable.Driver.main(Driver.java:17)
```

Whereas if I provide an absolute path, things work fine.  

```
java -jar target/sstable-tools-3.0.0-SNAPSHOT.jar toJson $PWD/ma-1-big-Data.db
```

Looks like a simple fix, will push something for this.